pkg/healthcheck/healthcheck.go

```diff
@@ -1623,6 +1624,29 @@ func (hc *HealthChecker) checkCertificatesConfig(ctx context.Context) (*tls.Cred
 	return issuerCreds, anchors, nil
 }
 
+// FetchCurrentConfiguration retrieves the current Linkerd configuration
+func FetchCurrentConfiguration(ctx context.Context, k kubernetes.Interface, controlPlaneNamespace string) (*corev1.ConfigMap, *l5dcharts.Values, error) {
+
+	// Get the linkerd-config values if present
+	configMap, configPb, err := FetchLinkerdConfigMap(ctx, k, controlPlaneNamespace)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if rawValues := configMap.Data["values"]; rawValues != "" {
+		var fullValues *l5dcharts.Values
```

Comment:
should `fullValues` just be a `l5dcharts.Values` instead of a pointer?  I assume that the default value for a pointer is nil, which isn't what we want.

Reply:
I'm testing this and it actually seems to work.  But I'm not sure why heh....